modules/effects/arrow-effects-data/src/main/kotlin/arrow/effects/typeclasses/MonadDefer.kt

```diff
@@ -44,6 +44,11 @@ interface MonadDefer<F> : MonadThrow<F>, Bracket<F, Throwable> {
 
   fun lazy(): Kind<F, Unit> = delay { }
 
+  fun <A> delayUnsafe(f: () -> Either<Throwable, A>): Kind<F, A> =
+    defer { f().fold({ raiseError<A>(it) }, { just(it) }) }
+
+  @Deprecated("Use delayUnsafe instead",
+          ReplaceWith("delayUnsafe(f)", "arrow.effects.typeclasses.MonadDefer"))
   fun <A> deferUnsafe(f: () -> Either<Throwable, A>): Kind<F, A> =
     defer { f().fold({ raiseError<A>(it) }, { just(it) }) }
```

Comment:
The implementation should just call `delayUnsafe`

Reply:
I implemented it like this so we do not have to copy the implementation when `deferUnsafe` gets deleted. But yes, it would be more clean to call `deferUnsafe` :+1: